### PR TITLE
Update to new impl package

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by adibbley** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% set name = "cuda-cccl" %}
 {% set version = "12.4.99" %}
-{% set impl_version = "2.0.1" %}
+{% set impl_version = "2.3.1" %}
```
Comment (bdice) [inline recipe/meta.yaml]: Maybe we should rename this:
```suggestion
{% set cccl_version = "2.3.1" %}
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -48,7 +48,7 @@ outputs:
       run:
         - {{ pin_compatible("cuda-version", max_pin="x.x") }}
         - cuda-cccl_{{ target_platform }} =={{ version }}
-        - cuda-cccl-impl =={{ impl_version }}
+        - cccl =={{ impl_version }}
```
Comment (bdice) [inline recipe/meta.yaml]:
```suggestion
        - cccl =={{ cccl_version }}
```

**Review by bdice — APPROVED** [review]
One rename suggestion, otherwise LGTM.

**Comment by jakirkham** [comment]
@kkraus14 and @jrhemstad could you please take a look?

**Review by jrhemstad — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

JFYI we are having issues with uploads atm: https://github.com/conda-forge/status/issues/174

So we will need to restart CI after that clears out

**Comment by jakirkham** [comment]
Restarted. Let's see how that goes

**Pull request merged**